ThirdPartyAdapters/mintegral/mintegral/src/main/java/com/google/ads/mediation/mintegral/rtb/MintegralRtbBannerAd.java

```
@@ -0,0 +1,195 @@
+
+
+  public MintegralRtbBannerAd(
+          @NonNull MediationBannerAdConfiguration mediationBannerAdConfiguration,
+          @NonNull MediationAdLoadCallback<MediationBannerAd, MediationBannerAdCallback>
+                  mediationAdLoadCallback) {
+    this.adConfiguration = mediationBannerAdConfiguration;
+    this.adLoadCallback = mediationAdLoadCallback;
+  }
+
+  /**
+   * Retrieves the bannerView pixels
+   *
+   * @param context the context object.
+   * @param dpValue the bannerView dip size
+   * @return the Mintegral bannerView pixels
+   */
+  private static int convertDpToPixels(Context context, float dpValue) {
+    final float scale = context.getResources().getDisplayMetrics().density;
+    return (int) (dpValue * scale + 0.5f);
+  }
+
+
+  private void initBannerAd() {
+    BannerSize bannerSize = null;
+    ArrayList<AdSize> supportedSizes = new ArrayList<>(3);
+    supportedSizes.add(new AdSize(320, 50));
+    supportedSizes.add(new AdSize(300, 250));
+    supportedSizes.add(new AdSize(728, 90));
+    AdSize closestSize = MediationUtils.findClosestSize(adConfiguration.getContext(), adConfiguration.getAdSize(), supportedSizes);
+    if (closestSize == null) {
+      AdError error = MintegralConstants.createAdapterError(
+              ERROR_BANNER_SIZE_MISMATCH,
+              "Failed to request banner ad from Mintegral. Invalid banner size.");
+      Log.w(TAG, error.toString());
+      adLoadCallback.onFailure(error);
+      return;
+    }
+
+    if (closestSize.equals(AdSize.BANNER)) { // 320 * 50
+      bannerSize = new BannerSize(BannerSize.STANDARD_TYPE, 0, 0);
+    }
+    if (closestSize.equals(AdSize.MEDIUM_RECTANGLE)) { // 300 * 250
+      bannerSize = new BannerSize(BannerSize.MEDIUM_TYPE, 0, 0);
+    }
+    if (closestSize.equals(AdSize.LEADERBOARD)) { // 728 * 90
+      bannerSize = new BannerSize(BannerSize.SMART_TYPE, closestSize.getWidth(), 0);
+    }
+    if (bannerSize == null) {
```

I think we should return a failure callback instead, publisher shouldn't set the banner sizes other than the ones above.

---

mintegral sdk supports it internally, and developers can customize the display size function, because different application developers may need different sizes